### PR TITLE
Feat #839 : Support for Editing User Details

### LINKED ISSUE #839: Support for editing user details including personal photo, client details
**Details :**

Whenever user clicks on his/her Personal photo either on the Dashboard or on the Navigation
Drawer, the user is taken to the his/her Profile page, there we can add Button, next to Photo (will
change present ImageView to circular ImageView, as current ImageView is too large and images
doesn’t look good there ). 

**Mockup :**

image

image

### PR DESCRIPTION
**Pull request by miPlodder**
Fixes #839 

pr839

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by miPlodder** [comment]
**Updated PR**
(I have not submitted change password, as password will change again and we have to change the password 4-5 times to keep the default-password "password". In GIF above, I have checked the API call.)

pr839_2
pr839_3

**Comment by miPlodder** [comment]
@therajanmaurya Please review this PR.

**Comment by miPlodder** [comment]
@therajanmaurya In this PR, username and clientname is stored incorrectly in SharedPrefs.

File: app/src/main/java/org/mifos/mobilebanking/presenters/UpdatePasswordPresenter.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import io.reactivex.observers.DisposableObserver;
+import io.reactivex.schedulers.Schedulers;
+import okhttp3.Credentials;
+import okhttp3.ResponseBody;
+
+public class UpdatePasswordPresenter extends BasePresenter<UpdatePasswordView> {
+
+    CompositeDisposable compositeDisposable;
+    DataManager dataManager;
+    PreferencesHelper preferencesHelper;
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/mobilebanking/presenters/UpdatePasswordPresenter.java]: Make above variables private

File: app/src/main/java/org/mifos/mobilebanking/presenters/UpdatePasswordPresenter.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    @Override
+    public void detachView() {
+        super.detachView();
+        compositeDisposable.clear();
+    }
+
+    public void updateAccountPassword(final UpdatePasswordPayload payload) {
+        getMvpView().showProgress();
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/mobilebanking/presenters/UpdatePasswordPresenter.java:55]: Check view is attached or not.

File: app/src/main/java/org/mifos/mobilebanking/presenters/UpdatePasswordPresenter.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+                    }
+
+                    @Override
+                    public void onComplete() {
+
+                    }
+                }));
+    }
+
+    public void createAuthenticationToken(String password) {
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/mobilebanking/presenters/UpdatePasswordPresenter.java]: Update method name `updateAuthenticationToken()`

File: app/src/main/java/org/mifos/mobilebanking/ui/fragments/UpdatePasswordFragment.java [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+    @Inject
+    UpdatePasswordPresenter presenter;
+
+    @Inject
+    PreferencesHelper preferencesHelper;
+
+    View rootView;
+    UpdatePasswordPayload payload;
+    boolean isFocusLostNewPassword = false;
+    boolean isFocusLostConfirmPassword = false;
```
Comment (therajanmaurya) [inline app/src/main/java/org/mifos/mobilebanking/ui/fragments/UpdatePasswordFragment.java]: Make above variables private

**Review by therajanmaurya — CHANGES_REQUESTED** [review]
@miPlodder Please make above-requested changes and fix that you mentioned username and clientname is stored incorrectly in SharedPrefs:

File: app/src/main/java/org/mifos/mobilebanking/presenters/HomePresenter.java [hunk 5]
```diff
@@ -89,8 +89,8 @@ public void onNext(Client client) {
                         getMvpView().hideProgress();
                         if (client != null) {
                             preferencesHelper.setOfficeName(client.getOfficeName());
-                            preferencesHelper.setUserName(client.getDisplayName());
-                            getMvpView().showUserDetails(preferencesHelper.getUserName());
+                            preferencesHelper.setClientName(client.getDisplayName());
```
Comment (miPlodder) [inline app/src/main/java/org/mifos/mobilebanking/presenters/HomePresenter.java:92]: @therajanmaurya These are the changes that I was talking of.

**Pull request merged**